tools/pythonpkg/src/pyconnection.cpp

```diff
@@ -402,6 +415,154 @@ shared_ptr<DuckDBPyConnection> DuckDBPyConnection::RegisterPythonObject(const st
 	return shared_from_this();
 }
 
+unique_ptr<DuckDBPyRelation> DuckDBPyConnection::ReadCSV(const string &name, const py::kwargs &kwargs) {
```

Comment:
Whats the reason you've implemented this using kwargs? Most of the type checking code could be deleted if you used explicitly stated arguments instead

Reply:
To copy the API of Pandas' read_csv function, if I didn't use kwargs they would have to be specified in the right order, no?

Reply:
You can provide keyword arguments in any order you like, it's only positional arguments that need to be provided in the right order (hence why they are positional)

Reply:
I'm not entirely sure what you have in mind to replace this

The keywords arguments are optional and don't have a default value

Reply:
Using `None` is exactly how you denote a parameter as being optional in Python actually.

You can have a look at the doc to see how they've defined it: https://pandas.pydata.org/docs/reference/api/pandas.read_csv.html

Unfortunately I haven't had the chance to complete, as I'm currently travelling, but you can look at the `Optional` class here for an example: https://github.com/duckdb/duckdb/pull/5732/files#diff-09d39950a22a7a86ba6b9b1fa29ab796c2d4a30196a64bb924f196cd8390a752